aten/src/ATen/native/native_functions.yaml

```diff
@@ -4359,12 +4359,12 @@
   use_c10_dispatcher: full
   variants: method, function
 
-- func: tril_indices(int row, int col, int offset=0, *, ScalarType? dtype=long, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
```

Comment:
Isn't it easier to have default values specified here rather somewhere in the code? native_functions.yaml is used as a reference doc often so it would make sense to have default types specified here.

Reply:
In an ideal world, sure. But not if it's at the cost of an extremely painful series of hacks in the codegen to support this. Better to just do something marginally worse (but still 100% functional) and then maybe fix it later when codegen can handle it.